### PR TITLE
feature(ui-ux): add themed and row component v2

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- Add themed component v2, i.e.: text, scrollview, view, touchable opacity, and section title
- Add number and text row v2

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/components/NumberRowV2.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+type INumberRowProps = React.PropsWithChildren<ViewProps> & NumberRowProps
+
+interface NumberRowProps extends ThemedProps {
+  lhs: NumberRowElement
+  rhs: rhsNumberRowElement
+  containerStyle?: ThemedProps & { style: ThemedProps & StyleProp<ViewStyle> }
+}
+
+interface rhsNumberRowElement extends NumberRowElement {
```
Comment (thedoublejay) [inline mobile-app/app/components/NumberRowV2.tsx:17]: interface shouldn't start with lowercase

File: mobile-app/app/components/themed/ThemedScrollViewV2.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useThemeContext } from '@shared-contexts/ThemeProvider'
+import { tailwind } from '@tailwind'
+import { forwardRef } from 'react'
+
+import { ScrollView } from 'react-native'
+import { ThemedProps } from './index'
+
+type ThemedScrollViewProps = ScrollView['props'] & ThemedProps
+
+export const ThemedScrollViewV2 = forwardRef(function (props: ThemedScrollViewProps, ref: React.Ref<any>): JSX.Element {
```
Comment (thedoublejay) [inline mobile-app/app/components/themed/ThemedScrollViewV2.tsx:10]: Why do we need a forwardref here?

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: mobile-app/app/components/themed/ThemedScrollViewV2.tsx (hunk 2 above)
Comment (kyleleow) [reply mobile-app/app/components/themed/ThemedScrollViewV2.tsx:10]: to allow scroll to top to work when tap on bottom tab menu

File: mobile-app/app/components/NumberRowV2.tsx (hunk 1 above)
Comment (kyleleow) [reply mobile-app/app/components/NumberRowV2.tsx]: updated

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**